MonoGame.Framework/Graphics/Vertices/VertexBuffer.cs

```diff
@@ -26,10 +26,6 @@ protected VertexBuffer(GraphicsDevice graphicsDevice, VertexDeclaration vertexDe
             this.VertexCount = vertexCount;
             this.BufferUsage = bufferUsage;
 
-            // Make sure the graphics device is assigned in the vertex declaration.
-            if (vertexDeclaration.GraphicsDevice != graphicsDevice)
-                vertexDeclaration.GraphicsDevice = graphicsDevice;

```

Comment:
Did some tests with XNA.  The `VertexDeclaration.GraphicsDevice` property remains `null` until a `VertexBuffer` is created using the vertex declaration.  After the `GraphicsDevice` is disposed, the vertex declaration retains the reference to the disposed `GraphicsDevice` object.  The disposed `GraphicsDevice` object is replaced with the new `GraphicsDevice` object when subsequently used again with a new `VertexBuffer`.